pandas/core/indexers/objects.py

```diff
@@ -66,9 +80,23 @@ def get_window_bounds(
 
         raise NotImplementedError
 
+    @Appender(get_window_bounds_doc)
+    def get_window_bounds2(
```

Comment:
Don't think this is a good idea. `get_window_bounds` should be modified instead

Reply:
This `get_window_bounds2` is indeed annoying and I'd also like a cleaner solution, if we can find one. The problem is that if changing the signature of `get_window_bounds` would break any existing user code that includes indexers derived from `BaseIndexer`. The immediate breakage shows up in the signature compatibility check in `BaseWindow._validate`. If we somehow relax this check then we'd need to add code in multiple places that branches on `get_window_bounds` returning either 2 (in legacy code) or 3 (in new code) arrays; this would also be annoying. Any idea how to get around this problem?

Reply:
Note that a similar issue arises with `homogeneous_func`, which now returns an additional array corresponding to `ref`. Here too, there is existing code of homogeneous functions that do not return such an array. In this case, I was able to encapsulate the issue.

Reply:
I think it's okay for `get_window_bounds` to have a new signature (new default kwarg `step`) and also relax the validation in `BaseWindow._validate`

Per my comment above, I think `get_window_bounds` can still return a tuple of 2 arrays?

Reply:
Yes, I believe it could return a tuple of 2 arrays when group-by is not a concern.

Reply:
yeah this is fine to break (mention in the whatsnew note is fine, but this is effectively developer api)

Reply:
@jreback: The documentation for [Series.rolling](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.rolling.html) and [DataFrame.rolling](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.rolling.html) seems to make this API user-facing. Are you sure it's OK to break?

Having said that, my current plan is to punt on group-by and other behaviors as suggested by @mroeschke, so the API breakage issue can be deferred on.